### PR TITLE
Modify APoT dequantize method

### PR DESCRIPTION
**Pull request by asl3**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82105
* __->__ #82126

### Summary
Modify APoT dequantize method to correctly add dequantized values to result numpy array and retain original tensor dimensions

### Test Plan
Run unit tests with: `python test/quantization/core/experimental/test_quantizer.py`

### CODE REVIEW

File: torch/ao/quantization/experimental/quantizer.py [hunk 1]
```diff
@@ -57,15 +57,15 @@ def quantize(self, tensor2quantize: Tensor):
...
-            np.append(result_temp, new_ele)
+        result_temp = np.empty(shape=apot_tensor_data.size())
+        for i in range(len(apot_tensor_data)):
+            new_ele = apot_to_float(apot_tensor_data[i], self.quantization_levels, self.level_indices)
+            result_temp[i] = new_ele
 
-        result = torch.from_numpy(result_temp).int()
+        result = torch.from_numpy(result_temp)
 
         return result
```
Comment (HDCharles) [inline torch/ao/quantization/experimental/quantizer.py:70]: seems like the output of this for a 3x3 tensor would be a 1x9 tensor, shouldn't this output a tensor with the same output dimensions as the input tensor?

can you add a unit test for this?

**Review by HDCharles — CHANGES_REQUESTED** [review]

File: torch/ao/quantization/experimental/quantizer.py (hunk 1 above)
Comment (asl3) [reply torch/ao/quantization/experimental/quantizer.py:75]: yes agreed, fixed!

File: torch/ao/quantization/experimental/quantizer.py [hunk 2]
```diff
@@ -57,15 +57,20 @@ def quantize(self, tensor2quantize: Tensor):
...
-            np.append(result_temp, new_ele)
+        result_temp = np.empty(shape=apot_tensor_data.size())
+        for i in range(len(apot_tensor_data)):
+            new_ele = apot_to_float(apot_tensor_data[i], self.quantization_levels, self.level_indices)
+            result_temp[i] = new_ele
+
+        result = torch.from_numpy(result_temp).reshape(orig_size)
 
-        result = torch.from_numpy(result_temp).int()
+        print(result)
```
Comment (HDCharles) [inline torch/ao/quantization/experimental/quantizer.py:73]: this should be removed

**Review by HDCharles — APPROVED** [review]
there's a lint error and a print still in the code

otherwise looks good

**Pull request closed without merging**